sql/errors.go

```diff
@@ -265,3 +260,46 @@ func CastSQLError(err error) (*mysql.SQLError, bool) {
 
 	return mysql.NewSQLError(code, sqlState, err.Error()), false
 }
+
+type UniqueKeyError struct {
```

Comment:
Nit. We can't make this look more like https://blog.golang.org/go1.13-errors ? If we're going to be adding test methods, etc.

Usage like

```go
errors.Is(err, &UniqueKeyError{})
```

or

```go
errors.Is(err, &UniqueKeyError{IsPK: true})
```

with

```go
var ue *UniqueKeyError
if errors.As(err &ue) {
    if ue.IsPK {
        ...
    } else {
        ...
    }
}
```

all looking reasonable. Or something?

Reply:
@reltuk I don't really want to be pulling in two packages named errors with incompatible functionality.  In gms we use the "gopkg.in/src-d/go-errors.v1" instead of the standard package "errors".  

An example incompatibility is go-errors packages define a Wrap method which isn't compatible with Unwrap defined in the standard errors package. 

I did rework this so it looks more like the rest of the error handling in this repo.  Thoughts?

Reply:
LGTM!

Reply:
@bheni we should sync up on this. I have another branch that's trying to remove these error definitions and use ErrDuplicateKey only